app/controllers/preferences_controller.rb

```diff
@@ -24,6 +28,15 @@ def create
     end
   end
 
+  def update
+    @preference = Preference.find(params[:id])
+    if @preference.update(permitted_params)
+      redirect_to preferences_path, notice: t('views.preferences.update_success')
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
```

Comment:
I would love to see some **feature specs** for the edit/update behavior. 
You can test that the form is render successfully, that the success message is shown when updating, that the attributes have been updated successfully, or even that when invalid params, test the model validations and that the error message are displayed.

Reply:
What you recommend would be great, it would help me avoid some mistakes. 
What do you think about implementing these tests in a new PR? Let me know if that's OK :smile:

Reply:
Sounds good!